### PR TITLE
Renaming Database with views error

### LINKED ISSUE #16422: Renaming Database with views throws an error
### Describe the bug

Renaming Database with views throws an error.

### To Reproduce

1. Go to a database with views.
2. Click on Operations
3. Rename the database
4. Click Go

### Expected behavior
The database should be renamed with views moved to the new database.

### Actual behavior
An error is thrown and only tables which are not view tables are moved.

### Screenshots
Screenshot from 2020-10-21 00-46-11
 The complete query is : ``` `CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY DEFINER VIEW `name_region`  AS SELECT `world_borders`.`NAME` AS `NAME`, `world_borders`.`REGION` AS `REGION` FROM `world_borders` WHERE (`world_borders`.`REGION` > 19)` ```

### Server configuration

- Operating system: Ubuntu 20
- Web server: PHP 7.4.3 Development Server 
- Database version:  MySQL 8.0.21
- PHP version: 7.4.3 
- phpMyAdmin version: master / QA_5_0

### Client configuration

- Browser: Firefox
- Operating system:Ubuntu 20

### Additional context

Relates: https://github.com/phpmyadmin/phpmyadmin/issues/16215

### PR DESCRIPTION
**Pull request by shucon**
### Description

Fixes: https://github.com/phpmyadmin/phpmyadmin/issues/16422

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]

**Comment by ibennetch** [comment]
Strangely, I can't reproduce this bug; nonetheless the fix looks good to me.

**Comment by shucon** [comment]
I think it only happens is specific conditions where the selected database is changed. Can you try please try using this sql file?
[opengis.zip](https://github.com/phpmyadmin/phpmyadmin/files/5423435/opengis.zip)

File: templates/database/operations/index.twig [hunk 1]
```diff
@@ -180,6 +180,11 @@
           </label>
         </div>
 
+        <div class="form-check">
+          <input class="form-check-input" type="checkbox" name="adjust_views" value="1" id="checkbox_views" checked>
+          <label class="form-check-label" for="checkbox_constraints">{% trans 'Adjust view definitions' %}</label>
```
Comment (williamdes) [inline templates/database/operations/index.twig:185]:
```suggestion
          <input class="form-check-input" type="checkbox" name="adjust_views" value="1" id="checkbox_views" checked>
          <label class="form-check-label" for="checkbox_views">{% trans 'Adjust view definitions' %}</label>
```
ID and for must be in sync

**Comment by williamdes** [comment]
Hello @shucon 
This is the second time I have a look to this pull-request and can not understand how a select database could solve a view adjustment ?
Do we already adjust views ?

**Comment by shucon** [comment]
> Hello @shucon
> This is the second time I have a look to this pull-request and can not understand how a select database could solve a view adjustment ?
> Do we already adjust views ?

Hey @williamdes the problem was that the `$dbi->query($sql_structure);` below the selectDb, queries on a different database if the database wasn't properly selected. Reason being that `$sql_structure` doesn't mention the Database to run on.

**Comment by williamdes** [comment]
> > Hello @shucon
> > This is the second time I have a look to this pull-request and can not understand how a select database could solve a view adjustment ?
> > Do we already adjust views ?
> 
> Hey @williamdes the problem was that the `$dbi->query($sql_structure);` below the selectDb, queries on a different database if the database wasn't properly selected. Reason being that `$sql_structure` doesn't mention the Database to run on.

Yup, this part is clear.
And what about the checkbox, why is it needed ?

**Comment by shucon** [comment]
> 
> Yup, this part is clear.
> And what about the checkbox, why is it needed ?

That's related to https://github.com/phpmyadmin/phpmyadmin/issues/16214 . Sorry if it's confusing. I fixed two problems in a single PR as they were related.

**Comment by williamdes** [comment]
> > Yup, this part is clear.
> > And what about the checkbox, why is it needed ?
> 
> That's related to #16214 . Sorry if it's confusing. I fixed two problems in a single PR as they were related.

Okay, but Isaac said "Checking that box would search and replace any view definitions that refer to the original database name and change them to refer to the newly copied database name." and I only can see a checkbox. This feels a little much like magic to me. Is there any code that does that somewhere I missed ?

**Comment by shucon** [comment]
> Okay, but Isaac said "Checking that box would search and replace any view definitions that refer to the original database name and change them to refer to the newly copied database name." and I only can see a checkbox. This feels a little much like magic to me. Is there any code that does that somewhere I missed ?

It's done here https://github.com/phpmyadmin/phpmyadmin/pull/16421/files . I mentioned it in that PR, forgot to mention it here. This PR is dependent on that one to be merged first.

**Comment by williamdes** [comment]
> It's done here #16421 . I mentioned it in that PR, forgot to mention it here. This PR is dependent on that one to be merged first.

Oh okay, would you mind moving the change into the other pull-request ?
- you remove the checkbox and post `or`
- I merge this one
- You rebase the other PR onto latest commit and add the checkbox

Does it seem possible for you ?
Just because I think I can merge this one quickly and the other one needs more deep review

**Comment by shucon** [comment]
Hey @williamdes, it's ready to be merged. I'll add the rest of the changes to the other PR.

**Comment by Lovor01** [comment]
Issue #16422 is closed, but not fixed; I have database with views, renaimng database does not copy views in new database and throws error.

**Comment by williamdes** [comment]
> Issue #16422 is closed, but not fixed; I have database with views, renaimng database does not copy views in new database and throws error. Phpmyadmin 5.2.0

Can you open a new issue please ?
Please add a screenshot if possible

**Comment by williamdes** [comment]
Done as  #17998

**Pull request merged**